src/test/scala/viper/gobra/GobraPackageTests.scala

```diff
@@ -22,58 +24,87 @@ import scala.concurrent.duration.Duration
 class GobraPackageTests extends GobraTests {
   override val testDirectories: Seq[String] = Vector("same_package")
 
+  override def buildTestInput(file: Path, prefix: String): DefaultAnnotatedTestInput = {
+    // get package clause of file and collect all other files belonging to this package:
+    val input = for {
+      pkgName <- getPackageClause(file.toFile)
+      currentDir = file.getParent.toFile
+      samePkgFiles = currentDir.listFiles
+        .filter(_.isFile)
+        .map(f => (f, getPackageClause(f)))
+        .filter { case (_, Some(clause)) if clause == pkgName => true }
+        .map { case (f, _) => f.toPath }
+        .sortBy(_.toString)
+        .toSeq
+    } yield DefaultTestInput(s"$prefix/$pkgName (${file.getFileName.toString})", prefix, samePkgFiles, Seq())
+    DefaultAnnotatedTestInput(input.get)
+  }
+
   override val gobraInstanceUnderTest: SystemUnderTest =
     new SystemUnderTest with TimingUtils {
       /** For filtering test annotations. Does not need to be unique. */
       override val projectInfo: ProjectInfo = new ProjectInfo(List("Gobra"))
 
       override def run(input: AnnotatedTestInput): Seq[AbstractOutput] = {
-        // extract package clause of input:
-        val resultOpt = for {
+        // test scallop parsing by giving package name and testing whether the same set of files is created
+        val currentDir = input.file.getParent
```

Comment:
`input.file` returns one of the (/the main) file of this test case (that potentially consists of multiple files) (`input` has basically two getters, `file` and `files` (probably for historical purposes))